### PR TITLE
Added more methods to `Pimcore/Model/Element/ElementInterface`

### LINKED ISSUE #12112: [Improvement]: Add more methods to ElementInterface
### Improvement description

Such as `setParentId()` `getParentId()`, `setParent()` ...

### PR DESCRIPTION
**Pull request by aashan10**
## Changes in this pull request  
Resolves #12112

### CODE REVIEW

**Review by dvesh3 — COMMENTED** [review]
can also add `setType($type)`..right?

File: models/Element/ElementInterface.php [hunk 1]
```diff
@@ -208,11 +206,23 @@ public function getLocked();
      */
     public function getParentId();
 
+    /**
+     * @param int|null $id
+     * @return static
```
Comment (blankse) [inline models/Element/ElementInterface.php]:
```suggestion
     *
     * @return $this
```

File: models/Element/ElementInterface.php [hunk 2]
```diff
@@ -208,11 +206,23 @@ public function getLocked();
...
+    public function setParentId($id);
+
     /**
      * @return self|null
      */
     public function getParent();
 
+    /**
+     * @param ElementInterface|null $parent
+     * @return static
```
Comment (blankse) [inline models/Element/ElementInterface.php]:
```suggestion
     *
     * @return $this
```

**Review by blankse — CHANGES_REQUESTED** [review]
The setters should return `$this`

**Comment by dvesh3** [comment]
@aashan10 please see https://github.com/pimcore/pimcore/pull/13290#pullrequestreview-1129863089

**Comment by aashan10** [comment]
Sure, I'll add the method.

**Comment by dvesh3** [comment]
@aashan10 please also add upgrade notes. thanks!

File: models/Element/ElementInterface.php [hunk 3]
```diff
@@ -208,11 +212,24 @@ public function getLocked();
...
```
Comment (jdreesen) [inline models/Element/ElementInterface.php]: Shouldn't this method return `$this`, too?

```suggestion
     *
     * @return $this
```

Comment (blankse) [inline models/Element/ElementInterface.php]:
```suggestion
     *
     * @return $this
```

Comment (blankse) [reply models/Element/ElementInterface.php]: You were a few seconds faster :)

**Pull request merged**